### ISSUE TITLE
`--split-bibdata` and `--semantic` options seem not to be working as expected

### ISSUE THREAD

**Issue by manuelfuenmayor** [metanorma/mnconvert#295]
When I apply `--split-bibdata` option, one asciidoc file is created with
the list of attributes inside along with a `.rxl` file. The rest of the document
content is dnot included.

On the other hand, when I apply `--semantic` option,  it creates the whole
Metanorma Asciidoc document, but not the XML version.

These options were tested on ISO 10303-24 from https://github.com/metanorma/iso-10303-detached-docs.

**Comment by Intelligent2013** [comment]
I'm a bit confused about option `--split-bibdata`.
This option was introduced here https://github.com/metanorma/mn2sts/issues/21. Initially, mnconvert (formerly sts2mn) generated one big adoc with bibdata, main sections and annexes. The option `--split-bibdata` was introduced  for generation of separated files for bibdata, each section, annex for STS to Metanorma Adoc conversion.
For STS to Metanorma XML conversion, this option allows to generate Relaton `.rxl` file (`bibdata` element from Metanorma XML).

Currently, mnconvert generates always standalone adocs for bibdata, each section and annex (I can't remember when this decision accepted). So, currently no make sense `--split-bibdata` for 'STS to Metanorma Adoc' conversion mode. I'll remove it.

Regarding `--split-bibdata` for 'STS to Metanorma XML conversion' mode, I think it should be renamed to `--relaton`. And in this case, the Relaton `.rxl` will be created additionally to full Metanorma XML.


`--semantic` make sense together with the option `--output-format xml` (STS XML to Metanorma XML conversion mode).
There isn't feature to generate Metanorma Adoc and XML simultaneously. You have to execute `mnconvert` twice.

**Comment by manuelfuenmayor** [comment]
> Regarding --split-bibdata for 'STS to Metanorma XML conversion' mode, I think it should be renamed to --relaton. And in this case, the Relaton .rxl will be created additionally to full Metanorma XML.

I have tried this configuration:
````
java -jar mnconvert/mnconvert-1.27.0.jar <XML_file>.xml -of xml -sb
````

It creates two files: `.rxl` and `.mn.xml`.
* `.rxl` only includes information about the converted document itself (i.e., it seems incomplete).
* `.mn.xml` file only includes the attributes of the document and nothing else (i.e., it is definitely incomplete).

This is the XML source I'm performing the tests on: 
[iso_std_iso_10303-28_ed-1_v1_en.zip](https://github.com/metanorma/mnconvert/files/10049836/iso_std_iso_10303-28_ed-1_v1_en.zip)

**Comment by manuelfuenmayor** [comment]
> --semantic make sense together with the option --output-format xml (STS XML to Metanorma XML conversion mode).
There isn't feature to generate Metanorma Adoc and XML simultaneously. You have to execute mnconvert twice.'

In this regard, I tried this configuration:
````
java -jar mnconvert/mnconvert-1.27.0.jar <XML_file>.xml -of xml --semantic
````
It creates an `.mn.xml` file that is pretty much like the conversion output from `-of xml` option.
In other words, `--semantic` option seems not to have any effect., in my opinion.

Can I ask what is the difference between Metanorma XML and Semantic Metanorma XML?

**Comment by Intelligent2013** [comment]
Sorry, I was wrong about `--split-bibdata`. When applied this option, the two files should be created:
- adoc with document attributes only;
- .rxl file with `bibdata` element only.

The full document doesn't generate.

**Comment by Intelligent2013** [comment]
> Can I ask what is the difference between Metanorma XML and Semantic Metanorma XML

Metanorma gems allow to generate two kind of XML files:
- semantic Metanorma XML (the parameter `-x xml` in `bundle exec metanorma -t iso -x xml document.adoc` for instance)
- presentation Metanorma XML (the parameter `-x presentation` in `bundle exec metanorma -t iso -x presentation document.adoc` for instance)

Semantic Metanorma XML is more structured than presentation XML (especially for terms, requirements). Presentation XML contains additional 'pre-rendered' information like clause's numbers, annex letters, clause's depth value, term's data ready to render, display order, note's/example's numbers, etc. (https://github.com/metanorma/mnconvert/issues/117, this list isn't complete yet). These pre-rendered information allows to render document in PDF/Word/HTML in the most unified forms. I.e. no need to 'calculate', for instance, clause's number in each PDF, Word, HTML processor separaterely.

**Comment by Intelligent2013** [comment]
> In other words, `--semantic` option seems not to have any effect., in my opinion.

@manuelfuenmayor as you can see at the screenshot below, there are a lot of differences between semantic vs. non-semantic (presentation) XML:
![image](https://user-images.githubusercontent.com/5559926/203593624-dd7a5176-bc90-4741-9dd5-ae31c254f50b.png)

(I use WinMerge + PrettifyXML unpacker plugin)

The issue with `--split-bibdata` fixed in mnconvert v1.28.
Now mnconvert generates .adoc and .rxl in both output-format adoc and xml.

**Comment by manuelfuenmayor** [comment]
Thanks @Intelligent2013 for your detailed explanation. Now, I understand the difference.

> The issue with --split-bibdata fixed in mnconvert v1.28.
Now mnconvert generates .adoc and .rxl in both output-format adoc and xml.

Ok, I will test it. Thanks!
